### PR TITLE
feat: git connect v2

### LINKED ISSUE #25588: Git connect modal ux improvements

### PR DESCRIPTION
**Pull request by brayn003**
## Description
UX improvements for Git Connect Flow
https://zpl.io/W4AQoek

#### PR fixes following issue(s)
Fixes #25588

#### Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/26725#issuecomment-1709723205
https://github.com/appsmithorg/appsmith/pull/26725#issuecomment-1711136694
>
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/GitSyncModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { useFeatureFlag } from "utils/hooks/useFeatureFlag";
+import GitSyncModalV1 from "./GitSyncModalV1";
+import GitSyncModalV2 from "./GitSyncModalV2";
+
+interface GitSyncModalProps {
+  isImport?: boolean;
+}
+
+function GitSyncModal(props: GitSyncModalProps) {
+  const isGitConnectV2Enabled = useFeatureFlag(
```
Comment (hetunandu) [inline app/client/src/pages/Editor/gitSync/GitSyncModal/index.tsx:12]: 💯

File: app/client/src/pages/Editor/gitSync/Tabs/GitSettings/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+// import type { AuthorInfo } from "./GitUserSettings";
+import GitUserSettings from "./GitUserSettings";
+import GitDisconnect from "./GitDisconnect";
+// import { getGlobalGitConfig } from "selectors/gitSyncSelectors";
+// import { useSelector } from "react-redux";
+
+function GitSettings() {
+  // const globalGitConfig = useSelector(getGlobalGitConfig);
```
Comment (hetunandu) [inline app/client/src/pages/Editor/gitSync/Tabs/GitSettings/index.tsx:9]: Remove comments if not needed

File: app/client/src/pages/Editor/gitSync/constants.ts [hunk 3]
```diff
@@ -51,3 +22,56 @@ export enum CREDENTIAL_MODE {
   MANUALLY = "MANUALLY",
   IMPORT_JSON = "IMPORT_JSON",
 }
+
+// import {
```
Comment (hetunandu) [inline app/client/src/pages/Editor/gitSync/constants.ts:26]: Why add commented code?

File: app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts [hunk 4]
```diff
@@ -6,6 +6,10 @@ import {
 import { useCallback, useEffect, useState } from "react";
 import { generateSSHKeyPair, getSSHKeyPair } from "actions/gitSyncActions";
 
+const NOOP = () => {
```
Comment (hetunandu) [inline app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts:9]: You can use lodash noop instead to creating one for each file
`_.noop`

**Review by hetunandu — CHANGES_REQUESTED** [review]
Minor changes needed

File: app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts (hunk 4 above)
Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts:9]: thanks, this makes life easier

**Comment by Parthvi12** [comment]
- [ ] @brayn003 Merge option doesn't show dropdown with all branches, can you please check:

https://github.com/appsmithorg/appsmith/assets/80334441/57a4805d-aa7a-4df9-b9a9-a7e99b57d710

**Comment by brayn003** [comment]
@Parthvi12 I have added the fixes. Can you please check again?

**Comment by brayn003** [comment]
> Screenshot 2023-09-08 at 11 46 36 AM

@Parthvi12  This is already an existing component, I have borrowed it from the old flow

**Comment by brayn003** [comment]
> > Screenshot 2023-09-08 at 11 46 36 AM
> 
> @Parthvi12 This is already an existing component, I have borrowed it from the old flow

Also, it is not currently possible to achieve this design using ADS input

**Comment by Parthvi12** [comment]
@brayn003 Thanks! Verified the fixes there are couple of issues which exist, can you please check that:

- [ ] When git connection fails, the error is thrown but can't see the error until we scroll up, also there is grammatical error in msg, adding recording:

https://github.com/appsmithorg/appsmith/assets/80334441/e40facd5-62d9-4e83-9e73-26d3087eb77a

- [ ] When we try to import empty repo, the connect modal doesn't throw any error:

Screenshot 2023-09-08 at 4 14 54 PM

- [ ] In Safari, the modal moves to top left for a moment and then come back to center

**Comment by brayn003** [comment]
> * [x]  In Safari, the modal moves to top left for a moment and then come back to center , when we open it.

Tested on Safari, unable to replicate. I have added a fix for this issue. Can you please recheck with the most recent DP

**Comment by Parthvi12** [comment]
@brayn003 Strange, Safari issue still happens for me, I will create a separate issue for it, not treating it as blocker for this PR


https://github.com/appsmithorg/appsmith/assets/80334441/4859fcef-d672-4a9a-8e1d-922d1a0dd4a4

File: app/client/cypress/e2e/Regression/ClientSide/Git/GitSync/GitConnectV2_spec.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    });
+
+    _.gitSync.CreateNConnectToGitV2();
+
+    cy.get("@gitRepoName").then((repName) => {
+      repoName = repName;
+    });
+  });
+
+  after(() => {
```
Comment (Parthvi12) [inline app/client/cypress/e2e/Regression/ClientSide/Git/GitSync/GitConnectV2_spec.ts:27]: @brayn003 Can we also add cypress tests for git import flow here, not a blocker for this PR.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**